src/main/java/core/basesyntax/service/impl/ProcessServiceCsvImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.constants.Activity;
+import core.basesyntax.service.ProcessService;
+import core.basesyntax.strategy.ActivitiesStrategy;
+import core.basesyntax.strategy.ActivitiesStrategyImpl;
+import core.basesyntax.strategy.handlers.ActivityHandler;
+import core.basesyntax.strategy.handlers.impl.BalanceActivityHandler;
+import core.basesyntax.strategy.handlers.impl.PurchaseActivityHandler;
+import core.basesyntax.strategy.handlers.impl.ReturnActivityHandler;
+import core.basesyntax.strategy.handlers.impl.SupplyActivityHandler;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class ProcessServiceCsvImpl implements ProcessService {
+    private static final String CSV_PUNCTUATION_MARK = ",";
+    private static final Integer CSV_ACTIVITY_POSITION = 0;
+    private static final Integer CSV_PRODUCT_POSITION = 1;
+    private static final Integer CSV_AMOUNT_POSITION = 2;
+    private final Map<String, ActivityHandler> activitiesStrategyMap = Map.of(
+            Activity.BALANCE.getActivity(), new BalanceActivityHandler(),
+            Activity.PURCHASE.getActivity(), new PurchaseActivityHandler(),
+            Activity.RETURN.getActivity(), new ReturnActivityHandler(),
+            Activity.SUPPLY.getActivity(), new SupplyActivityHandler()
+            );
+    private final ActivitiesStrategy activitiesStrategy =
+            new ActivitiesStrategyImpl(activitiesStrategyMap);
+    private final DataLineProcessService dataLineProcessService =
+            new DataLineProcessService();
```

Comment:
The code should not be implementation dependent
All non-constant fields should be initialized in the constructor
```suggestion
    public ProcessServiceCsvImpl(ActivitiesStrategy activitiesStrategy, DataLineProcessService dataLineProcessService) {
    this.activitiesStrategy = activitiesStrategy;
    this.dataLineProcessService = dataLineProcessService;
    }
```